src/goto-instrument/contracts/contracts.h

```diff
@@ -56,6 +58,7 @@ class code_contractst
 public:
   code_contractst(goto_modelt &goto_model, messaget &log)
     : ns(goto_model.symbol_table),
+      goto_model(goto_model),
       symbol_table(goto_model.symbol_table),
       goto_functions(goto_model.goto_functions),
```

Comment:
Having goto_model, symbol_table and goto_functions is redundant.
Either have goto_model or symbol_table+goto_functions. I'd prefer the former.

Reply:
Ah, ok, I see. You are just using them as shortcuts for the goto_model members. That's fine.

Reply:
But then I wonder: is it really useful to have these at class level?

Reply:
Yes, might still be confusing that changing one thing has side effects on the other.

Reply:
Hi! I had to add the goto_model as a class member because the nondet static call I needed to add requires a goto_model. I also think that exploding the goto_model into shortcuts to its members is confusing as it could make you think these members are disconnected in some way. But it also seems like this style of having shortcuts to the symbol table, namespace of function maps like this occurs in several places in the codebase.
Do you want me to do a cleanup pass and remove shortcuts ?

Reply:
Yes, I would suggest that we clean out these particular short cuts. Note that the situation is a bit different from the widespread use of a namespace member for there is no risk of confusion as to whether a namespace is different or the same as the member of another class member. (In other words: the current code raises the question whether `.symbol_table` is the same as `.goto_model.symbol_table` or not.)